.github/workflows/pr.yml

```diff
@@ -0,0 +1,32 @@
+name: Validate PR
```

Comment:
Just thinking if the action name should be 
Build and Test. - Runs for all branches.
Publish - Runs for main.

Reply:
updated the name